diff -urN dmidecode-2.5.orig/Makefile dmidecode-2.5/Makefile
--- dmidecode-2.5.orig/Makefile	2004-11-11 22:23:34.453338320 +0100
+++ dmidecode-2.5/Makefile	2004-11-11 22:40:56.207967544 +0100
@@ -25,7 +25,7 @@
 DESTDIR =
 prefix  = /usr/local
 sbindir = $(prefix)/sbin
-mandir  = $(prefix)/man
+mandir  = $(prefix)/share/man
 man8dir = $(mandir)/man8
 
 all : dmidecode biosdecode ownership vpddecode
