--- Makefile~	2004-02-23 19:56:04.000000000 +0100
+++ Makefile	2004-03-25 18:49:15.000000000 +0100
@@ -71,10 +71,10 @@
 	install -m 755 biosdecode $(PREFIX)/sbin
 	install -m 755 ownership $(PREFIX)/sbin
 	install -m 755 vpddecode $(PREFIX)/sbin
-	install -m 644 man/dmidecode.8 $(PREFIX)/man/man8
-	install -m 644 man/biosdecode.8 $(PREFIX)/man/man8
-	install -m 644 man/ownership.8 $(PREFIX)/man/man8
-	install -m 644 man/vpddecode.8 $(PREFIX)/man/man8
+	install -m 644 man/dmidecode.8 $(PREFIX)/share/man/man8
+	install -m 644 man/biosdecode.8 $(PREFIX)/share/man/man8
+	install -m 644 man/ownership.8 $(PREFIX)/share/man/man8
+	install -m 644 man/vpddecode.8 $(PREFIX)/share/man/man8
 
 uninstall :
 	rm -f $(PREFIX)/sbin/dmidecode $(PREFIX)/man/man8/dmidecode.8
